### PR TITLE
[MASSEMBLY-974] Drop deprecated repository element

### PR DESCRIPTION
**Pull request by cstamas**
As a followup for deprecation MASSEMBLY-957 in version 3.4.0, the deprecated element is being dropped.

As a consequence, this removes deprecated MAT as well.

--

https://issues.apache.org/jira/browse/MASSEMBLY-974

### CODE REVIEW

**Comment by cstamas** [comment]
On my workstation one IT fails, the `projects/reproducible`, unsure why. Failure is:
```
assert content == effective
       |       |  |
       |       |  40755 dir-A/
       |       |  40755 dir-C/
       |       |  40755 dir-b/
       |       |  40755 dir-b/B2/
       |       |  40755 dir-b/B4/
       |       |  40755 dir-b/b1/
       |       |  40755 dir-b/b3/
       |       |  40755 dir-d/
       |       |  100664 Uppercase.txt
       |       |  100664 dir-A/A2.txt
       |       |  100664 dir-A/A4.txt
       |       |  100664 dir-A/a1.txt
       |       |  100664 dir-A/a3.txt
       |       |  100664 dir-C/C.txt
       |       |  100664 dir-b/B2/B2.txt
       |       |  100664 dir-b/B4/B4.txt
       |       |  100664 dir-b/b.txt
       |       |  100664 dir-b/b1/b1.txt
       |       |  100664 dir-b/b3/b3.txt
       |       |  100664 dir-d/d.txt
       |       |  100775 executable.txt
       |       |  100664 lowercase.txt
       |       |  2456cb51390a059bf9b93cba4a0e0fc8f6bcf038 reproducible-1.0-src.zip.sha1
       |       |  2437692cda28c00d3c34936104ead651a6a348e1 reproducible-1.0-src.jar.sha1
       |       |  c6a6ab2d2a9bc040ecb5925ee2df4b5ec63194d0 reproducible-1.0-src.tar.sha1
       |       false
       40775 dir-A/
       40775 dir-C/
       40775 dir-b/
       40775 dir-b/B2/
       40775 dir-b/B4/
       40775 dir-b/b1/
       40775 dir-b/b3/
       40775 dir-d/
       100664 Uppercase.txt
       100664 dir-A/A2.txt
       100664 dir-A/A4.txt
       100664 dir-A/a1.txt
       100664 dir-A/a3.txt
       100664 dir-C/C.txt
       100664 dir-b/B2/B2.txt
       100664 dir-b/B4/B4.txt
       100664 dir-b/b.txt
       100664 dir-b/b1/b1.txt
       100664 dir-b/b3/b3.txt
       100664 dir-d/d.txt
       100775 executable.txt
       100664 lowercase.txt
       50116502c6107740c2a35ef296b5abda08c5dec7 reproducible-1.0-src.zip.sha1
       cc7e3a984179f63d6b37bc86c61e9cc461c62288 reproducible-1.0-src.jar.sha1
       3efc10ec9c3099ba061e58d5b2a935ba643da237 reproducible-1.0-src.tar.sha1
```

File: src/main/mdo/assembly.mdo [hunk 1]
```diff
@@ -226,7 +226,7 @@
         </field>
         <field>
           <name>repositories</name>
-          <version>1.0.0+</version>
```
Comment (cstamas) [inline src/main/mdo/assembly.mdo:229]: This was 1.0.0 and unintentional change (copy+paste) happened, this should be `1.0.0/2.1.1`.

File: src/main/mdo/assembly.mdo [hunk 2]
```diff
@@ -226,7 +226,7 @@
         </field>
         <field>
           <name>repositories</name>
```
Comment (slawekjaranowski) [inline src/main/mdo/assembly.mdo:228]: Can we remove also `repositories` from model as we remove it from code?

Comment (cstamas) [reply src/main/mdo/assembly.mdo:228]: Ah, did I miss range from class? As field is gone, that's sure

Comment (cstamas) [reply src/main/mdo/assembly.mdo:228]: removed

**Review by slawekjaranowski — APPROVED** [review]

**Pull request merged**